From f4ecba77ec1eb0576b8731c749336ea3889eba73 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Andreas=20K=2E=20H=C3=BCttel?= <dilfridge@gentoo.org>
Date: Sun, 13 Oct 2019 05:25:28 +0200
Subject: [PATCH 5/6] Gentoo: ld: enable new dtags by default for linux/gnu
 targets

Original author: Mike Frysinger <vapier@gentoo.org>

The "new" dtags options have been around for 14+ years now, so for Linux
and GNU targets, enable them by default.

2012-01-21  Mike Frysinger  <vapier@gentoo.org>

        * emultempl/elf32.em (gld${EMULATION_NAME}_before_parse): Set
        link_info.new_dtags to TRUE for linux/gnu targets.
        * NEWS: Mention new dtags default.

2013-01-22  Roland McGrath  <mcgrathr@google.com>

        * emultempl/elf32.em (gld${EMULATION_NAME}_before_parse): Set
        new_dtags to TRUE for *-*-nacl* targets.
---
 ld/emultempl/elf.em | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/ld/emultempl/elf.em b/ld/emultempl/elf.em
index bfaf8130a3e..c4c42b7371a 100644
--- a/ld/emultempl/elf.em
+++ b/ld/emultempl/elf.em
@@ -81,6 +81,16 @@ gld${EMULATION_NAME}_before_parse (void)
   input_flags.dynamic = ${DYNAMIC_LINK-true};
   config.has_shared = `if test -n "$GENERATE_SHLIB_SCRIPT" ; then echo true ; else echo false ; fi`;
   config.separate_code = `if test "x${SEPARATE_CODE}" = xyes ; then echo true ; else echo false ; fi`;
+EOF
+
+case ${target} in
+  *-*-linux-* | *-*-k*bsd*-* | *-*-gnu* | *-*-nacl*)
+    fragment <<EOF
+  link_info.new_dtags = true;
+EOF
+    ;;
+esac
+fragment <<EOF
   link_info.check_relocs_after_open_input = true;
 EOF
 if test -n "$COMMONPAGESIZE"; then
-- 
2.31.1

